bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/handler/FreeAtHomeDeviceHandler.java

```diff
@@ -0,0 +1,608 @@
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.thing.type.ChannelKind;
+import org.openhab.core.thing.type.ChannelType;
+import org.openhab.core.thing.type.ChannelTypeBuilder;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.StateDescriptionFragmentBuilder;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.FrameworkUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link FreeAtHomeDeviceHandler} is responsible for handling the generic free@home device main communication
+ * and thing updates
+ *
+ * @author Andras Uhrin - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class FreeAtHomeDeviceHandler extends BaseThingHandler implements FreeAtHomeDeviceStateListener {
+
+    private static final String CHANNEL_URI = "channel-type:freeathomesystem:config";
+
+    private final Logger logger = LoggerFactory.getLogger(FreeAtHomeDeviceHandler.class);
+    private FreeAtHomeDeviceDescription device = new FreeAtHomeDeviceDescription();
+    private FreeAtHomeChannelTypeProvider channelTypeProvider;
+    private TranslationProvider i18nProvider;
+    private LocaleProvider localeProvider;
+    private Locale locale;
+    private Bundle bundle;
+
+    private Map<ChannelUID, FreeAtHomeDatapointGroup> mapChannelUID = new HashMap<ChannelUID, FreeAtHomeDatapointGroup>();
+    private Map<String, ChannelUID> mapEventToChannelUID = new HashMap<String, ChannelUID>();
+
+    public FreeAtHomeDeviceHandler(Thing thing, FreeAtHomeChannelTypeProvider channelTypeProvider,
+            TranslationProvider i18nProvider, LocaleProvider localeProvider) {
+        super(thing);
+
+        this.channelTypeProvider = channelTypeProvider;
+        this.i18nProvider = i18nProvider;
+        this.localeProvider = localeProvider;
+        this.bundle = FrameworkUtil.getBundle(getClass());
+        this.locale = localeProvider.getLocale();
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+
+        scheduler.execute(() -> {
+            FreeAtHomeBridgeHandler freeAtHomeBridge = null;
+            Map<String, String> properties = getThing().getProperties();
+            Bridge bridge = this.getBridge();
+
+            logger.debug("Start creating device - device id: {}", device.getDeviceId());
+
+            if (bridge != null) {
+                ThingHandler handler = bridge.getHandler();
+
+                if (handler instanceof FreeAtHomeBridgeHandler bridgeHandler) {
+                    freeAtHomeBridge = bridgeHandler;
```

Comment:
**Fixed**

Reply:
I'm afraid there is a slight misunderstanding here. What I meant was that the variable `freeAtHomeBridge` is not needed, but the `instanceof` check was fine since it provided the variable `bridgeHandler` guaranteed to be the correct type, which can then be used directly. It seems this check was now removed, and the local variable `freeAtHomeBridge` is left unused rather than being removed.

Reply:
**Fixed**

it shall be OK now